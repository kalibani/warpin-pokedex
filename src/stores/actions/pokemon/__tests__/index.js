import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
// Actions to be tested
import * as pokemonActions from '../index';

const mockStore = configureStore([thunk]);
const store = mockStore();
describe('pokemonActions', () => {
  beforeEach(() => {
    store.clearActions();
  });

  describe('setLoading', () => {
    test('Dispatches the correct action and payload', () => {
      const expectedActions = [
        {
          'payload': true,
          'type': 'pokemon/SET_LOADING',
        },
      ];

      store.dispatch(pokemonActions.setLoading(true));
      expect(store.getActions()).toMatchSnapshot();
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  describe('setDataPokemon', () => {
    test('Dispatches the correct action and payload', () => {
      const expectedActions = [
        {
          'payload': [],
          'type': 'pokemon/SET_DATA_POKEMON',
        },
      ];

      store.dispatch(pokemonActions.setDataPokemon([]));
      expect(store.getActions()).toMatchSnapshot();
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  describe('setNextUrl', () => {
    test('Dispatches the correct action and payload', () => {
      const expectedActions = [
        {
          'payload': '',
          'type': 'pokemon/SET_NEXT_URL',
        },
      ];
      
      store.dispatch(pokemonActions.setNextUrl(''));
      expect(store.getActions()).toMatchSnapshot();
      expect(store.getActions()).toEqual(expectedActions);
    });
  });


});