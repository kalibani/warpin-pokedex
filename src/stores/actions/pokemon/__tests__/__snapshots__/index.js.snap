// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemonActions setDataPokemon Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Array [],
    "type": "pokemon/SET_DATA_POKEMON",
  },
]
`;

exports[`pokemonActions setLoading Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": true,
    "type": "pokemon/SET_LOADING",
  },
]
`;

exports[`pokemonActions setNextUrl Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": "",
    "type": "pokemon/SET_NEXT_URL",
  },
]
`;
